pkg/devfile/adapters/docker/utils/utils.go

```diff
@@ -9,18 +9,48 @@ import (
 	"github.com/docker/docker/api/types"
 	"github.com/docker/docker/api/types/container"
 	"github.com/docker/docker/api/types/mount"
+
+	adaptersCommon "github.com/openshift/odo/pkg/devfile/adapters/common"
 	"github.com/openshift/odo/pkg/devfile/parser/data/common"
 	"github.com/openshift/odo/pkg/lclient"
+	"github.com/openshift/odo/pkg/log"
+	"github.com/openshift/odo/pkg/util"
+
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+)
+
+const (
+	supervisordVolume = "supervisord"
+	projectsVolume    = "projects"
 )
 
 // ComponentExists checks if Docker containers labeled with the specified component name exists
 func ComponentExists(client lclient.Client, name string) bool {
+	containers := GetComponentContainers(client, name)
+	return len(containers) != 0
+}
+
+// GetComponentContainers returns a list of the running component containers
+func GetComponentContainers(client lclient.Client, name string) (containers []types.Container) {
 	containerList, err := client.GetContainerList()
 	if err != nil {
-		return false
+		return
```

Comment:
`return nil` ? what are we returning? we should be explicit

Reply:
I have defined the func as `func GetComponentContainers(client lclient.Client, name string) (containers []types.Container) {`; so just a `return` would return the empty `containers []types.Container` in this place